Makefile

```diff
@@ -12,6 +12,53 @@ ARCH            := $(shell uname -m)
 # Default target
 all: build test
 
+#
+# OpenAPI Code Generation
+#
+OPENAPI_SPEC := api/openapi/owner-server.yaml
+GENERATED_DIR := generated
+
+.PHONY: validate-openapi
+validate-openapi:
+	@echo "Validating OpenAPI specification..."
+	@command -v openapi-generator >/dev/null 2>&1 || { \
+		echo "Error: openapi-generator CLI not found. Please install it first:"; \
+		echo "  npm install @openapitools/openapi-generator-cli -g"; \
+		echo "  or"; \
+		echo "  brew install openapi-generator"; \
+		exit 1; \
+	}
+	openapi-generator validate -i $(OPENAPI_SPEC)
+
+.PHONY: generate-api
+generate-api: validate-openapi
+	@echo "Generating Go server and client code from OpenAPI specification..."
+	@mkdir -p $(GENERATED_DIR)/server $(GENERATED_DIR)/client
+	openapi-generator generate \
+		-i $(OPENAPI_SPEC) \
+		-g go-server \
+		-o $(GENERATED_DIR)/server \
+		--additional-properties=packageName=openapi,outputAsLibrary=true,sourceFolder=src/main/go
+	openapi-generator generate \
+		-i $(OPENAPI_SPEC) \
+		-g go \
+		-o $(GENERATED_DIR)/client \
```

Comment:
Instead of using a Makefile target to generate the api I would prefer to use the golang native approach to generate code, please see https://github.com/oapi-codegen/oapi-codegen/?tab=readme-ov-file#for-go-124 as an example. We will need a Makefile target anyway but in this case it will run `go generate` and we will make the `build` target to depend on it.

This leads me to open the discussion about using the openapi generator tools (java) or the more native `oapi-codegen` shown in the link above.

After thinking a lot about this I think we should use the native tool as it's very well documented and used by other projects, but as I said this is open to discussion.